splash/tests/test_execute.py

```diff
@@ -2504,6 +2519,66 @@ def test_get(self):
         self.assertStatusCode(resp, 200)
         self.assertEqual(JsRender.template, resp.text)
 
+    def test_get_with_default_headers(self):
+        resp = self.request_lua("""
+        function main(splash)
+            response = nil
+            splash:on_response(function (res)
+                response = res
+            end)
+            assert(splash:http_get(splash.args.url))
+            return response.request.headers
+        end
+        """, {"url": self.mockurl("jsrender")})
+        self.assertStatusCode(resp, 200)
+        headers = resp.json()
+        self.assertNotEqual(len(headers), 0)
+        self.assertEqual(len(headers), 1)
+        self.assertIn("Mozilla/5.0", headers["user-agent"])
+
+    def test_get_with_custom_headers(self):
+        resp = self.request_lua("""
+        function main(splash)
+            splash:set_custom_headers({
+                ["Header-1"] = "Value 1",
+                ["Header-2"] = "Value 2",
+                })
+            response = assert(splash:http_get(splash.args.url))
+            return response.request.headers
+        end
+        """, {"url": self.mockurl("jsrender")})
+        self.assertStatusCode(resp, 200)
+        headers = resp.json()
+        self.assertNotEqual(len(headers), 0)
+        self.assertEqual(headers["Header-1"], "Value 1")
+        self.assertEqual(headers["Header-2"], "Value 2")
+        self.assertIn("user-agent", headers)
+
+    def test_get_with_custom_ua(self):
+        resp = self.request_lua("""
+        function main(splash)
+            splash:set_user_agent("CUSTOM UA")
+            response = assert(splash:http_get(splash.args.url))
+            return response.request.headers
+        end
+        """, {"url": self.mockurl("jsrender")})
+        self.assertStatusCode(resp, 200)
+        headers = resp.json()
+        self.assertNotEqual(len(headers), 0)
+        self.assertEqual(headers["user-agent"], "CUSTOM UA")
+
+    def test_get_with_custom_ua_in_headers(self):
+        resp = self.request_lua("""
+        function main(splash)
+            response = assert(splash:http_get{splash.args.url, headers={["user-agent"]="Value 1"}})
+            return response.request.headers
+        end
```

Comment:
I added tests for UA for requests used when rendering. Turns out UA was not set when there was 'headers' argument passed because of some minor bug comparing string with bytes. Now only need to differentiate behavior of headers argument regarding go() and http_get